packages/widget/src/components/amount-selector/amount-selector.ts

```diff
@@ -1,117 +1,147 @@
 import type { Resource } from '@buildwithsygma/sygma-sdk-core';
 import type { HTMLTemplateResult } from 'lit';
 import { html } from 'lit';
-import { customElement, property, query } from 'lit/decorators.js';
+import { customElement, property, query, state } from 'lit/decorators.js';
 import { ifDefined } from 'lit/directives/if-defined.js';
-import { map } from 'lit/directives/map.js';
 import { when } from 'lit/directives/when.js';
+import { classMap } from 'lit/directives/class-map.js';
 import { BaseComponent } from '../base-component/base-component';
+import type { DropdownOption } from '../internal/dropdown/dropdown';
+import { networkIconsMap } from '../../assets';
 import { styles } from './styles';
 
 @customElement('sygma-resource-selector')
 export class AmountSelector extends BaseComponent {
   static styles = styles;
+
   @property({
     type: Array,
     hasChanged: (n, o) => n !== o
   })
   resources: Resource[] = [];
 
-  @property({
-    type: Boolean
-  })
+  @property({ type: Boolean })
   disabled = false;
 
-  @property({
-    type: String
-  })
+  @property({ type: String })
   accountBalance?: string;
 
-  @property({
-    type: String
-  })
+  @property({ type: String })
   preselectedToken?: string;
 
-  @property({
-    type: Number
-  })
+  @property({ type: Number })
   preselectedAmount?: number;
 
-  @property({
-    attribute: false
-  })
+  @property({ attribute: false })
   onResourceSelected?: (resource: Resource) => void;
 
-  @property({
-    attribute: false
-  })
+  @property({ attribute: false })
   onAmountChange?: (amount: number) => void;
 
   @query('.amountSelectorInput', true)
   _input!: HTMLInputElement;
 
-  useMaxBalance = (): void => {
-    this.preselectedAmount = Number.parseFloat(this.accountBalance!);
-    this._onInputAmountChange();
+  @state() validationMessage: string | null = null;
+
+  _useMaxBalance = (): void => {
+    this._input.value = this.accountBalance!;
+    this._validateAmount(this._input.value);
   };
 
   _onInputAmountChange = (): void => {
-    const amount = Number.parseFloat(this._input.value);
-    this.onAmountChange?.(amount);
+    const amount = this._input.value;
+    if (this._validateAmount(amount)) {
+      this.onAmountChange?.(Number.parseFloat(amount));
+    }
   };
 
-  _onResourceSelected = (event: Event): void => {
-    const { value } = event.target as HTMLOptionElement;
-    const resource = this.resources.find((n) => String(n.resourceId) == value);
-    if (resource) {
-      this.onResourceSelected?.(resource);
-    }
+  _onResourceSelected = (option: DropdownOption): void => {
+    const resource = this.resources.find(
+      (resource) => String(resource.resourceId) == option.id
+    );
+    if (resource) this.onResourceSelected?.(resource);
   };
 
-  renderBalance(): HTMLTemplateResult {
+  _validateAmount(amount: string): boolean {
```

Comment:
does it make sense to you to move this function to the utils folder? Wdyt?

Reply:
I don't think it's reusable now as the main purpose is to set the error message. But we can think about it later if we would need to use it in more places